catalog/solr/catalog-solr-offline-gazetteer/src/test/groovy/ddf/catalog/solr/offlinegazetteer/GazetteerQueryOfflineSolrSpec.groovy

```diff
@@ -0,0 +1,344 @@
+    def "test queryById exception"() {
+        setup:
+        1 * solrClient.query(*_) >> { throw new SolrServerException("solr exception") }
+        when:
+        GeoEntry result = testedClass.queryById("test")
+
+        then:
+        GeoEntryQueryException e = thrown()
+    }
+
+    def "test getSuggestedNames normal"() {
+        setup:
+        int maxResults = 10
+        1 * solrClient.query(*_) >> { SolrQuery query ->
+            assert query.get("suggest.count") == "${maxResults}"
+            assert query.requestHandler == "/suggest"
+            Mock(QueryResponse) {
+                getSuggesterResponse() >> Mock(SuggesterResponse) {
+                    getSuggestions() >>
+                            [(GazetteerQueryOfflineSolr.SUGGEST_PLACE_KEY): [Mock(Suggestion) {
+                                getPayload() >> "id"
+                                getTerm() >> "title"
+
+                            }]]
+                }
+            }
+        }
+        when:
+        List<Suggestion> results = testedClass.getSuggestedNames("place", maxResults)
+
+        then:
+        results.size() == 1
+    }
+
+    def "test getSuggestedNames maxResults"() {
+        setup:
+        int maxResults = 101
+        1 * solrClient.query(*_) >> { SolrQuery query ->
+            assert query.get("suggest.count") == "${GazetteerQueryOfflineSolr.MAX_RESULTS}"
+            assert query.requestHandler == "/suggest"
+            Mock(QueryResponse) {
+                getSuggesterResponse() >> Mock(SuggesterResponse) {
+                    getSuggestions() >>
+                            [(GazetteerQueryOfflineSolr.SUGGEST_PLACE_KEY): [Mock(Suggestion) {
+                                getPayload() >> "id"
+                                getTerm() >> "title"
+
+                            }]]
+                }
+            }
+        }
+        when:
+        List<Suggestion> results = testedClass.getSuggestedNames("place", maxResults)
+
+        then:
+        results.size() == 1
+    }
+
+    def "test getSuggestedNames solrclient exception"() {
+        setup:
+        int maxResults = 10
+        1 * solrClient.query(*_) >> { throw new SolrServerException("exception") }
+
+        when:
+        List<Suggestion> results = testedClass.getSuggestedNames("place", maxResults)
+
+        then:
+        GeoEntryQueryException e = thrown()
+    }
+
+    def "getNearestCities"() {
+        setup:
+        String locationWkt = "POINT (-98.86253 29.18968)"
+        String pointAbout42kmAway = "POINT (-98.496708 29.418376)"
+        int radiusInKm = 50
+        int maxResults = 10
+        1 * solrClient.query(*_) >> { SolrQuery query, METHOD method ->
+            assert METHOD.POST == method
+
+
+            // TODO (RCZ) - I really don't like extracting the polygon from the solr query. its
+            // risky logic and closely coupling to the implementation. But I also don't want to
+            // open access and have a @VisibleForTesting..
```

Comment:
to resolve -- any reviewers opinions on this?  you can see the specific logic inside of `extractIntersectsPolygon`

Reply:
Done